### PR TITLE
Shared config threshold logic.

### PR DESCRIPTION
**Pull request by bpblanken**
Ticket: #3642 

Screenshot 2023-10-26 at 12 10 54 PM

### CODE REVIEW

File: ui/shared/components/panel/search/constants.js [hunk 1]
```diff
@@ -472,17 +473,12 @@ export const IN_SILICO_FIELDS = [REQUIRE_SCORE_FIELD, ...PREDICTOR_FIELDS.filter
...
-            <div>{`Red > ${dangerThreshold}`}</div>
-            <div>{`Yellow > ${warningThreshold}`}</div>
-          </div>
-        )}
+        {thresholds && predictorColorRanges(thresholds)}
       </div>
     )
     return {
       labelHelp,
+      popupProps: thresholds ? { hoverable: true } : {},
```
Comment (bpblanken) [inline ui/shared/components/panel/search/constants.js:481]: `hoverable` makes the link clickable.

File: ui/shared/components/panel/search/constants.js [hunk 2]
```diff
@@ -472,17 +473,12 @@ export const IN_SILICO_FIELDS = [REQUIRE_SCORE_FIELD, ...PREDICTOR_FIELDS.filter
     const labelHelp = (
       <div>
         {`Enter a numeric cutoff for ${label}`}
-        {dangerThreshold && (
```
Comment (hanars) [inline ui/shared/components/panel/search/constants.js:475]: 🤦

File: ui/shared/components/panel/search/constants.js (hunk 1 above)
Comment (hanars) [reply ui/shared/components/panel/search/constants.js:481]: I think its fine to just always make the help label popup `hoverable`, I don't think theres any real advantage to it not being anywhere else

File: ui/shared/utils/constants.js [hunk 3]
```diff
@@ -1343,6 +1344,36 @@ export const PREDICTOR_FIELDS = [
   { field: 'mitotip', indicatorMap: MITOTIP_MAP },
   { field: 'hmtvar', thresholds: [undefined, undefined, 0.35, 0.35, undefined] },
 ]
+export const PRED_COLOR_MAP = ['green', 'olive', 'grey', 'yellow', 'red', '#8b0000']
```
Comment (hanars) [inline ui/shared/utils/constants.js:1347]: the name of this constant is okay for something used internally in a file, but if we are going to make it an importable constant we should probably name it something more decriptive and accurate, maybe `PREDICTION_COLOR_LIST` or `PREDICTION_COLORS`? Also, maybe it makes more sense not to export it at all and instead move all the helper functions that use it into this file, but thats up to you

Comment (bpblanken) [reply ui/shared/utils/constants.js]: I moved the function using `PRED_COLOR_MAP` into here.  Seemed cleaner!

**Review by hanars — APPROVED** [review]

**Pull request merged**